### PR TITLE
Go 56 fix fork detector

### PR DESCRIPTION
**Pull request by alexeykiselev**

### CODE REVIEW

File: cmd/forkdetector/internal/storage.go [hunk 1]
```diff
@@ -0,0 +1,547 @@
...
+	blockLinksPrefix                // Keys to store the block link by the block number
+	heightsPrefix                   // Keys to store the signatures of blocks by its heights
+	peerLeashPrefix                 // Keys to store the numbers of blocks last seen from peers
+	peerNodePrefix                  // Keys to store peers by its IPs
+)
+
+var (
+	blocksCounterKey = []byte{blocksCounterPrefix}
+	zeroSignature    = crypto.Signature{}
+	maxSignature     = crypto.Signature{0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff}
```
Comment (zer0main) [inline cmd/forkdetector/internal/storage.go]: `maxSignature = crypto.Signature{bytes.Repeat([]byte{0xff}, crypto.SignatureSize)}` looks nicer.

File: cmd/forkdetector/internal/types_test.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+			Version:     tc.version,
+			Attempts:    tc.attempts,
+			NextAttempt: tc.next,
+			State:       tc.state,
+		}
+		assert.Equal(t, tc.exp, pn.String())
+	}
+}
+
+//func TestPeerDesignationString(t *testing.T) {
```
Comment (zer0main) [inline cmd/forkdetector/internal/types_test.go]: Commented tests still have to be commented?

File: cmd/forkdetector/internal/storage.go (hunk 1 above)
Comment (alexeykiselev) [reply cmd/forkdetector/internal/storage.go]: Won't work. Here we have to initialise array, not slice. Using repeat we have to convert generated slice of bytes into array, which in turn could return an error.

File: cmd/forkdetector/internal/types_test.go (hunk 2 above)
Comment (alexeykiselev) [reply cmd/forkdetector/internal/types_test.go]: My bad

File: cmd/forkdetector/internal/api.go [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+		http.Error(w, fmt.Sprintf("Invalid block signature: %v", err), http.StatusBadRequest)
+		return
+	}
+	block, ok, err := a.storage.block(sig)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Failed to complete request: %v", err), http.StatusInternalServerError)
+		return
+	}
+	if !ok {
+		http.Error(w, "Block not found", http.StatusNotFound)
```
Comment (Frozen) [inline cmd/forkdetector/internal/api.go]: return

File: cmd/forkdetector/internal/dispatcher.go [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+	defer func(conn *Conn) {
+		d.schedule.remove(conn)
+	}(conn)
+	d.schedule.append(conn)
+	err := conn.DialAndServe(addr.String())
+	if err != nil {
+		zap.S().Errorf("Failed to establish connection with %s: %v", addr.String(), err)
+		err := d.registry.PeerDiscarded(addr)
+		if err != nil {
+
```
Comment (Frozen) [inline cmd/forkdetector/internal/dispatcher.go]: empty

**Pull request merged**